pkg/apis/projectcalico/v3/nodestatus.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	KindCalicoNodeStatus     = "CalicoNodeStatus"
+	KindCalicoNodeStatusList = "CalicoNodeStatusList"
+)
+
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CalicoNodeStatusList is a list of CalicoNodeStatus resources.
+type CalicoNodeStatusList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Items []CalicoNodeStatus `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type CalicoNodeStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   CalicoNodeStatusSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status CalicoNodeStatusStatus `json:"status,omitempty" protobuf:"bytes,2,opt,name=status"`
+}
+
+// CalicoNodeStatusSpec contains the specification for a CalicoNodeStatus resource.
+type CalicoNodeStatusSpec struct {
+	// The node name identifies the Calico node instance for node status.
+	Node string `json:"node,omitempty" validate:"required,name"`
+
+	// Classes specifies type of information CalicoNodeStatus should contain.
```

Comment:
This should have a more detailed user facing comment.

e.g., "Classes declares the types of information to monitor for this calico/node, and allows for selective status reporting about certain subsets of information."

It might be better to rename this field to something more obivous, as well.

e.g.,

```yaml
monitorStatusFor:
- Agent
- BGP
- Routes
```

or 

```yaml
reportStatusFor:
```

Reply:
I'll update the comment but I still feel `Classes` is a good name for the field.